### PR TITLE
MAT-4645: add shared logic for measures

### PR DESCRIPTION
**Pull request by chubert-sb**
## MADiE PR

Jira Ticket: [MAT-4645](https://jira.cms.gov/browse/MAT-4645)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [x] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [x] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [x] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/components/editMeasure/measureDetails/measureInformation/MeasureInformation.tsx [hunk 1]
```diff
@@ -146,7 +146,12 @@ export default function MeasureInformation() {
       await handleSubmit(values),
   });
 
-  const canEdit = measure?.createdBy === userName;
+  const isOwner = measure?.createdBy === userName;
+  const canEdit =
+    measure?.createdBy === userName ||
```
Comment (jkotanchik-SB) [inline src/components/editMeasure/measureDetails/measureInformation/MeasureInformation.tsx]: I just realized the left side of the OR can be replaced with `isOwner`.

File: src/components/editMeasure/measureDetails/stewardAndDevelopers/StewardAndDevelopers.test.tsx [hunk 2]
```diff
@@ -39,37 +55,44 @@ const mockMetaData = {
...
+  useKeyPress: jest.fn(() => false),
   measureStore: {
     updateMeasure: jest.fn((measure) => measure),
-    state: null,
-    initialState: null,
+    state: jest.fn().mockImplementation(() => mockMeasure),
+    initialState: jest.fn().mockImplementation(() => null),
     subscribe: (set) => {
-      set(mockMeasure);
+      // set(mockMeasure)
```
Comment (jkotanchik-SB) [inline src/components/editMeasure/measureDetails/stewardAndDevelopers/StewardAndDevelopers.test.tsx]: Re-enable or remove dead code, pls.

File: src/components/editMeasure/measureDetails/stewardAndDevelopers/StewardAndDevelopers.test.tsx [hunk 3]
```diff
@@ -147,17 +170,40 @@ describe("Steward and Developers component", () => {
   });
 
   it("should disable dropdowns if the user is not the owner of the measure", async () => {
```
Comment (jkotanchik-SB) [inline src/components/editMeasure/measureDetails/stewardAndDevelopers/StewardAndDevelopers.test.tsx]: Recommend renaming as we're verifying edit permissions.
```suggestion
  it("should disable dropdowns if the user does not have measure edit permissions", async () => {
```

File: src/components/editMeasure/measureDetails/measureMetadata/MeasureMetadata.test.tsx [hunk 4]
```diff
@@ -272,7 +273,7 @@ describe("MeasureRationale component", () => {
     });
   });
 
-  it("Should have no input field if user is not the measure owner", () => {
+  it("Should have no Save button if user is not the measure owner", () => {
```
Comment (jkotanchik-SB) [inline src/components/editMeasure/measureDetails/measureMetadata/MeasureMetadata.test.tsx]:
```suggestion
  it("Should have no Save button if the user cannot edit the measure", () => {
```

File: src/components/editMeasure/measureDetails/measureMetadata/MeasureMetadata.test.tsx [hunk 5]
```diff
@@ -282,7 +283,7 @@ describe("MeasureRationale component", () => {
     expect(saveButton).not.toBeInTheDocument();
   });
 
-  it("test - Should have no Save button if user is not the measure owner", () => {
+  it("test - Should have no input field if user is not the measure owner", () => {
```
Comment (jkotanchik-SB) [inline src/components/editMeasure/measureDetails/measureMetadata/MeasureMetadata.test.tsx]:
```suggestion
  it("test - Should have no input field if user cannot edit the measure", () => {
```

**Review by jkotanchik-SB — DISMISSED** [review]

File: package-lock.json [hunk 6]
```diff
@@ -111,6 +111,21 @@
         "webpack-merge": "^5.8.0"
       }
     },
+    "../../libraries/madie-models": {
+      "name": "@madie/madie-models",
+      "version": "1.2.8",
+      "license": "ISC",
```
Comment (jkotanchik-SB) [inline package-lock.json]: The local config got committed. I think the only way to get it to point at the npm version is to re-run `npm install`.

Comment (chubert-sb) [reply package-lock.json]: I think i got this fixed

File: src/components/editMeasure/measureDetails/measureMetadata/MeasureMetadata.test.tsx (hunk 5 above)
Comment (chubert-sb) [reply src/components/editMeasure/measureDetails/measureMetadata/MeasureMetadata.test.tsx]: fixed

File: src/components/editMeasure/measureDetails/measureMetadata/MeasureMetadata.test.tsx (hunk 4 above)
Comment (chubert-sb) [reply src/components/editMeasure/measureDetails/measureMetadata/MeasureMetadata.test.tsx]: fixed

File: src/components/editMeasure/measureDetails/stewardAndDevelopers/StewardAndDevelopers.test.tsx (hunk 3 above)
Comment (chubert-sb) [reply src/components/editMeasure/measureDetails/stewardAndDevelopers/StewardAndDevelopers.test.tsx]: fixed

File: src/components/editMeasure/measureDetails/stewardAndDevelopers/StewardAndDevelopers.test.tsx (hunk 2 above)
Comment (chubert-sb) [reply src/components/editMeasure/measureDetails/stewardAndDevelopers/StewardAndDevelopers.test.tsx]: fixed

File: src/components/editMeasure/measureDetails/measureInformation/MeasureInformation.tsx (hunk 1 above)
Comment (chubert-sb) [reply src/components/editMeasure/measureDetails/measureInformation/MeasureInformation.tsx]: fixed

**Review by jkotanchik-SB — DISMISSED** [review]

**Review by jkotanchik-SB — APPROVED** [review]

**Pull request merged**